packages/ember-application/lib/system/application.js

```diff
@@ -693,9 +693,15 @@ var Application = Namespace.extend(DeferredMixin, {
     var namespace = this;
     var initializer;
 
+    var makeGraphValue = function(initializer) {
+      return function() {
+        return initializer.initialize.apply(initializer, arguments);
```

Comment:
@rwjblue We can't simply call the function, we need to use either `call` or `apply`. I can update the PR to use `call` with named arguments if you would prefer?